packages/shared/sierra-client/src/patron.ts

```diff
@@ -74,25 +76,92 @@ function getPatronNameNonMarc(content: string): PatronName {
   };
 }
 
-interface VarField {
-  fieldTag: string,
-  content: string,
-  subfields: VarSubField[]
+// There's a bunch of hardcoded fields in here. This is intentional, these are static and don't change and mostly
+// indicate that the user has self-registered. I don't think we need to extract them out into configuration.
+export function toCreatePatron(title: string, firstName: string, lastName: string, pin: string): PatronCreate {
+  return {
+    pin: pin,
+    pMessage: 's',
+    homeLibraryCode: 'sreg',
+    patronType: 29,
+    fixedFields: {
+      46: {
+        label: "USER CAT.",
+        value: "13"
+      },
+    },
+    varFields: [
+      {
+        fieldTag: "n",
+        marcTag: "100",
+        subfields: [
+          {
+            tag: "a",
+            content: lastName
+          },
+          {
+            tag: "c",
+            content: title
+          },
+          {
+            tag: "b",
+            content: firstName
+          }
+        ]
+      }
+    ]
+  }
 }
 
-interface VarSubField {
-  tag: string,
-  content: string
+// The Patron record creation endpoint returns in response to a successful record creation the full URL to the API
+// endpoint to query that new record. We're not interested in that, just give us the Patron record number.
+export function extractRecordNumberFromLink(link: string): number {
+  const match = link.match(/^https:\/\/.+?\/v6\/patrons\/(\d+)$/);
+  if (!match || match.length < 2) {
+    throw new Error('Patron creation link [' + link + '] not in expected format');
+  }
+  return Number(match[1]);
 }
 
-interface PatronName {
+export interface PatronRecord {
+  recordNumber: number;
+  barcode: string;
   title: string;
   firstName: string;
   lastName: string;
+  email: string;
 }
 
-export interface PatronRecord extends PatronName {
-  recordNumber: number;
-  barcode: string;
-  emailAddress: string;
+// This represents the data required to create a Patron record in Sierra. The 'fixedFields' a bit odd, as the keys of
+// the embedded objects appear to be array indices of some description, but 'fixedFields' itself isn't an array...?
+export interface PatronCreate {
+  pin: string,
+  pMessage: string,
+  homeLibraryCode: string,
+  patronType: number,
+  fixedFields: {
+    46: {
+      label: string,
+      value: string
+    }
+  },
+  varFields: {
+    fieldTag: string,
+    marcTag: string,
+    subfields: {
+      tag: string,
+      content: string
+    }[]
+  }[]
+}
+
+interface VarField {
+  fieldTag: string,
+  content?: string,
+  subfields?: SubField[]
+}
+
+interface SubField {
+  tag: string,
+  content: string
```

Comment:
Are these types ever used? I can't find any references to them.

Reply:
[Here](https://github.com/wellcomecollection/identity/blob/feat/account_registration_backend_support/packages/shared/sierra-client/src/patron.ts#L1), [here](https://github.com/wellcomecollection/identity/blob/feat/account_registration_backend_support/packages/shared/sierra-client/src/patron.ts#L10) and [here](https://github.com/wellcomecollection/identity/blob/feat/account_registration_backend_support/packages/shared/sierra-client/src/patron.ts#L33).